### PR TITLE
Adding session02 homework

### PR DESCRIPTION
**Pull request by meshmote**
Pull request is from master because I tried out the merge function, then kept doing session02 homework in the newly merged master.  I'll keep the next HW on a session03 branch.  Can't keep this stuff straight.  :-)

Thanks!
Robert

### CODE REVIEW

**Comment by PythonCHB** [comment]
No problem with the branches -- and If I can keep up with you, it's not a big deal to use branches for this anyway...

-Chris

File: Students/RPerkins/session02/ack.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    assert ack(2, 2) == 7
+    assert ack(2, 3) == 9
+    assert ack(2, 4) == 11
+    assert ack(3, 0) == 5
+    assert ack(3, 1) == 13
+    assert ack(3, 2) == 29
+    assert ack(3, 3) == 61
+    assert ack(3, 4) == 125
+    #  max recursion depth exceeded after this point
+    #assert ack(4, 0) == 2**2**2-3
```
Comment (PythonCHB) [inline Students/RPerkins/session02/ack.py:41]: nice! where did you find that?

File: Students/RPerkins/session02/series.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+
+def fibonacci(n):
+    """Return the 'n'th value in the Fibonacci series"""
+
+# test for input in valid range
+    if n < 0:
+        return None
+
+# test for n == 0
```
Comment (PythonCHB) [inline Students/RPerkins/session02/series.py:11]: this is exactly the kind of not-very-useful comment you shouldn't bother with ;-)

File: Students/RPerkins/session02/series.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    if n == 1:
+        return 1
+    else:
+        return lucas(n-1) + lucas(n-2)
+
+def sum_series(n,first=0,second=1):
+    """Return the 'n'th value of a variable series"""
+
+    """The first two numbers are passed via "first" and "second"
+    and each subsequent element is the sum of the 'n-1'th and 'n-2'th element (yes, i said tooth)"""
```
Comment (PythonCHB) [inline Students/RPerkins/session02/series.py:42]: :-)

File: Students/RPerkins/session02/ack.py (hunk 1 above)
Comment (meshmote) [inline Students/RPerkins/session02/ack.py:41]: I thought the HW instructions said to go up to m=4, n=4 on the table in the
Wikipedia article, so I started out with an assert statement for each cell
in the table (the table gave the higher order answers as powers of two).
The max recursion depth error message was thrown by the interpreter on the
first run, so I just commented them out one by one till it stopped
complaining.

On Sun, Oct 12, 2014 at 9:27 PM, Christopher H.Barker, PhD <
notifications@github.com> wrote:

> In Students/RPerkins/session02/ack.py:
> 
> > -    assert ack(1, 1) == 3
> > -    assert ack(1, 2) == 4
> > -    assert ack(1, 3) == 5
> > -    assert ack(1, 4) == 6
> > -    assert ack(2, 0) == 3
> > -    assert ack(2, 1) == 5
> > -    assert ack(2, 2) == 7
> > -    assert ack(2, 3) == 9
> > -    assert ack(2, 4) == 11
> > -    assert ack(3, 0) == 5
> > -    assert ack(3, 1) == 13
> > -    assert ack(3, 2) == 29
> > -    assert ack(3, 3) == 61
> > -    assert ack(3, 4) == 125
> > -    #  max recursion depth exceeded after this point
> > -    #assert ack(4, 0) == 2**2**2-3
> 
> nice! where did you find that?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/UWPCE-PythonCert/IntroToPython/pull/12/files#r18753604
> .

## 

_Robert Perkins_
_Mobile:_ 425-239-9269
_Email:_ meshmote@gmail.com
 _Professional Profile
http://www.linkedin.com/pub/robert-perkins/9/a6a/a91_
http://maps.google.com/maps?q=&hl=en
See who we know in common
http://www.linkedin.com/e/wwk/29505421/?hs=false&tok=0zCreXIp0dhRQ1Want a
signature like this?
http://www.linkedin.com/e/sig/29505421/?hs=false&tok=1-_hPrm6YdhRQ1

Comment (PythonCHB) [inline Students/RPerkins/session02/ack.py:41]: On Sun, Oct 12, 2014 at 9:36 PM, Robert Perkins notifications@github.com
wrote:

> I thought the HW instructions said to go up to m=4, n=4 on the table in
> the Wikipedia article,
> 
> yup -- I wanted you to hit that recursion error... :-)
> 
> so I started out with an assert statement for each cell in the table (the
> table gave the higher order answers as powers of two).
> 
> ahh -- I had forgotten those where there.

Nice job.

-Chris

## 

Christopher Barker, PhD

Python Language Consulting
- Teaching
- Scientific Software Development
- Desktop GUI and Web Development
- wxPython, numpy, scipy, Cython

**Pull request merged**